### DISCUSSION TITLE
Нарушенная сортировка у модуля Library

### DISCUSSION THREAD

**Discussion by amd64fox** [Chimildic/goofy#178] [Помощь с ошибкой]
При работе с модулем  `Library` нарушается сортировка массива.

Обычный пример

Беру треки из плейлиста и делаю из них лайки.
```javascript
let love_tracks = Source.getPlaylistTracks('name', 'id');
Library.saveFavoriteTracks(love_tracks);
```
Получаю хаос в сортировке. Пробовал использовать  `Order.sort`  и сортировал по ключу `added_at`, результат нулевой.
Такие же проблемы заметил с `Library.followArtists` и `Library.saveAlbums`

**Comment by Chimildic** [comment]
Запросы отправляются группами по 50 элементов. Как следствие у них одинаковая дата добавления. Поэтому сортировка ломается.

**Reply by amd64fox** [reply]
Так это можно исправить или такой запрос не просто так сделан ?

**Reply by Chimildic** [reply]
Можно отправлять по одному элементу (возможно добавив задержку). Но это потратит много запросов (1 к 1) и увеличит время выполнения.
```js
  tracks.forEach(t => {
    Library.saveFavoriteTracks([t]);
    // Utilities.sleep(2);
  });
```

**Reply by amd64fox** [reply]
Для скрипта "Резервная копия аккаунта" конечно такой вариант все лимиты порвет.
